### PR TITLE
Stop supporting Python 3.6

### PR DESCRIPTION
**Pull request by nikfilippas**
As mentioned in #949 , `setuptools` won't work anymore because it requires py37+.
Python 3.6 is past its end of life so with this PR we stop supporting it and require that py37+ is used, in order to install CCL.

### CODE REVIEW

**Comment by damonge** [comment]
Yeah, we may want to hear people out a bit before we actually do this.

**Comment by nikfilippas** [comment]
CCL won't install if you don't have py37+ anyway.

**Comment by damonge** [comment]
ah, right. OK, let's do it then.

**Comment by hsinfan1996** [comment]
Using `setuptools_scm` v6.x solves #949 I think. And I can install CCL from source with python 3.6.13 in a conda environment. Is there anything wrong with my setup?

**Comment by hsinfan1996** [comment]
I actually don't mind bumping up the minimum python version. Just don't think #949 is an unsolvable issue.

**Comment by nikfilippas** [comment]
`setuptools_scm=7.0.0` was released on June 21st this year. Earlier versions required `py36+` but v7 requires `py37+`.
The problem is that if users have updated their `setuptools` installation they won't be able to install CCL. I'm sure we can patch this, but we'd need to check which `setuptools` version every user has, and I think this is too much of a hassle for little benefit since `py36` has reached its end of life a long time ago.

**Comment by nikfilippas** [comment]
An additional hurdle is that `setuptools` is the very first module loaded when installing CCL, so I think there isn't an easy way to force-install the older version in case the user has `setuptools v7+` installed. We list the package requirements *with* that very package!

File: setup.py [hunk 1]
```diff
@@ -92,6 +92,7 @@ def run(self):
     include_package_data=True,
     use_scm_version=True,
     distclass=Distribution,
+    python_requires='>3.7.0',
```
Comment (hsinfan1996) [inline setup.py:95]: Do you need '>=3.7.0' instead of '>3.7.0' here?

Comment (nikfilippas) [reply setup.py:95]: yes! thank you!

**Review by damonge — APPROVED** [review]

**Pull request merged**